portal-web/docroot/html/taglib/ui/icon/link_content.jspf

```diff
@@ -42,11 +42,19 @@
 	</c:otherwise>
 </c:choose>
 
+<%
+String getLocalizeMessage = localizeMessage ? LanguageUtil.get(resourceBundle, message) : message;
+%>
+
 <c:choose>
 	<c:when test="<%= (iconMenuIconCount != null) && ((iconMenuSingleIcon == null) || iconMenuShowWhenSingleIcon) %>">
-		<span class="taglib-text-icon"><%= localizeMessage ? LanguageUtil.get(resourceBundle, message) : message %></span>
+		<c:if test="<%= Validator.isNotNull(getLocalizeMessage) %>">
+			<span class="taglib-text-icon"><%= getLocalizeMessage %></span>
```

Comment:
I'm not 100% sure about this one. In the issue you mention `Elements in the DOM with class hide-accessible with null value inside` but this is certainly not the case. Removing this text will likely have visible effects and maybe cause regressions. Could you elaborate on why this is needed?

Reply:
Hi @jbalsas,

I will try to explain, the client is mostly worried about the accessibility point of view, if the screen readers tools would read this null value.
And the other problem they face after changing their theme to display the classes hide-accessible they could see the null value on some cases when hovering on some elements:
![image](https://user-images.githubusercontent.com/59687465/117289365-cc2af380-ae6c-11eb-9180-0c15fc7ee479.png)

Maybe my approach to not displaying this element when null is wrong, and we should just put an empty text instead.

I hope these helps.

Let me know what you think.
Regards

Reply:
Hey @georgel-pop-lr, let me try to clarify what I meant.

```java
localizeMessage ? LanguageUtil.get(resourceBundle, message) : message
```

The only way this returns `null` is if `message` is `null` as you've already verified. Thus, I'm not sure it makes sense to add the null check on the outer expression, but rather check before we even attempt to call `LanguageUtil.get`.

> And the other problem they face after changing their theme to display the classes hide-accessible they could see the null value on some cases when hovering on some elements:

If that's the case, they would also be seeing non-null values. Is this what they're trying to achieve?

> Maybe my approach to not displaying this element when null is wrong, and we should just put an empty text instead.

I think that's okay, it might make sense to show nothing if there's no content to show. You are touching 2 different elements, one that has the `hide-accessible` class and one that does not. In the ticket you only refer to the one with the class, so I'm wondering why also touch the one without it. Have you been able to validate the effects of that change in some occurrence?